### PR TITLE
Fix Distributed Load Retry Logic

### PR DESCRIPTION
**Pull request by bradyoo**
Hypothetical Environment:

3 nodes, each with a job worker and a worker.
Suppose a file x.txt has 10 blocks to load but one of its blocks (**Block 1**) is in the middle of being read on (**Worker A**), Attempting to load this file creates 10 tasks and the task to load **Block 1** is assigned to be in **Worker A**.

Previous behavior:

The task in an attempt to load **Block 1** in **Worker A** fails. This fails the job but the other 9 tasks are still continuing. The client notices the failure and triggers a retry. Retry notices that none of the 10 blocks are still loaded yet and shuffles the location of where they are getting loaded.

New behavior:

The task in an attempt to load **Block 1** in **Worker A** fails. This fails the job but the other 9 tasks are still continuing. The client waits for all of its 10 tasks to finish despite knowing that the job has already failed. After all the tasks are complete/failed, it reruns the job again. This time, the server knows that 9 of its blocks are already loaded and only one more needs to get loaded.

### CODE REVIEW

**Comment by bradyoo** [comment]
fix seems to be promising @calvinjia

**Comment by bradyoo** [comment]
Tests in error: 
alluxio.server.configuration.ConfigCheckerIntegrationTest.multiNodes(alluxio.server.configuration.ConfigCheckerIntegrationTest)
  Run 1: ConfigCheckerIntegrationTest.multiNodes:142 » Timeout Timed out waiting for a ...
  Run 2: ConfigCheckerIntegrationTest.multiNodes:142 » Timeout Timed out waiting for a ...

alluxio.server.ft.ZookeeperFailureIntegrationTest.zkConnectionPolicy_Session(alluxio.server.ft.ZookeeperFailureIntegrationTest)
  Run 1: ZookeeperFailureIntegrationTest.zkConnectionPolicy_Session:139 » Timeout Timed...
  Run 2: ZookeeperFailureIntegrationTest.zkConnectionPolicy_Session:139 » Timeout Timed...

alluxio.server.ft.journal.MultiMasterJournalTest.testTtl(alluxio.server.ft.journal.MultiMasterJournalTest)
  Run 1: MultiMasterJournalTest.before:53 » IO java.util.concurrent.TimeoutException: T...
  Run 2: MultiMasterJournalTest.before:53 » IO java.util.concurrent.TimeoutException: T...

alluxio.server.ft.journal.MultiProcessCheckpointTest.test(alluxio.server.ft.journal.MultiProcessCheckpointTest)
  Run 1: MultiProcessCheckpointTest.test:63 » Timeout Timed out waiting for all nodes r...
  Run 2: MultiProcessCheckpointTest.test:63 » Timeout Timed out waiting for all nodes r...

**Comment by bradyoo** [comment]
We have way too many flaky tests...

**Comment by bradyoo** [comment]
alluxio-bot, test this please.

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 1]
```diff
@@ -89,7 +88,8 @@ private boolean run() {
         }
         return true;
       }
-      LOG.warn("Failed to complete job after retries: {}", mJobConfig);
+      System.out.println(String.format("Failed to complete loading %s after retries.",
+          mJobConfig.getFilePath()));
```
Comment (calvinjia) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: Print number of retries?

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 2]
```diff
@@ -111,7 +111,27 @@ private Status check() {
...
+        }
+      }
+
+      if (finished) {
+        if (jobInfo.getStatus().equals(Status.FAILED)) {
+          System.out.println(String.format("Attempt %s to load %s failed because: %s",
+              mRetryPolicy.getAttemptCount(), mJobConfig.getFilePath(),
+              jobInfo.getErrorMessage()));
+        } else if (jobInfo.getStatus().equals(Status.COMPLETED)) {
+          System.out.println(String.format("Attempt %s to load %s completed",
```
Comment (calvinjia) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: Perhaps word this a bit differently, `Successfully loaded path %s after %d attempts`? Also, is it possible for this to become spammy if a large number of files is to be loaded?

**Comment by bradyoo** [comment]
@calvinjia: feedbacks addressed. I agree with the concern for spamminess but for now I think this is fine. The code was also previously printing all of the starts and successes.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by bradyoo** [comment]
Can you cherry-pick this for me to 2.1? @calvinjia

**Comment by bradyoo** [comment]
Can you cherry-pick this to 2.1.2? @calvinjia

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please.

**Pull request merged**